File: android-sdk/src/main/kotlin/com/example/featureflagsdk/android/FeatureEvaluation.kt
package com.example.featureflagsdk.android

/**
 * Result returned by the backend for a single feature flag evaluation.
 *
 * @param flagKey         The flag identifier.
 * @param enabled         Whether the flag is active for the given context.
 * @param reasonCode      Why the flag resolved to this value (e.g. RULE_MATCH, FLAG_DISABLED).
 * @param matchedRuleId   ID of the rule that produced the result, if any.
 * @param snapshotVersion Version of the snapshot used during evaluation.
 * @param releaseKey      Release tag associated with the flag config.
 */
data class FeatureEvaluation(
    val flagKey: String,
    val enabled: Boolean,
    val reasonCode: String,
    val matchedRuleId: String? = null,
    val snapshotVersion: Long,
    val releaseKey: String? = null,
)

File: ios-sdk/Sources/FeatureFlagSDK/FeatureEvaluation.swift
import Foundation

/// Result returned by the backend for a single feature flag evaluation.
public struct FeatureEvaluation: Decodable {
    /// The flag identifier.
    public let flagKey: String
    /// Whether the flag is active for the given context.
    public let enabled: Bool
    /// Why the flag resolved to this value (e.g. RULE_MATCH, FLAG_DISABLED).
    public let reasonCode: String
    /// ID of the rule that produced the result, if any.
    public let matchedRuleId: String?
    /// Version of the snapshot used during evaluation.
    public let snapshotVersion: Int
    /// Release tag associated with the flag config.
    public let releaseKey: String?
}
